### PR TITLE
[docs] blue-green deploys with pinned schemas

### PR DESCRIPTION
**Pull request by shorgi**
Deploy previews:
- [Overview](https://deploy-preview-2850--apollo-federation-docs.netlify.app/managed-federation/overview)
- [Deploy variant](https://deploy-preview-2850--apollo-federation-docs.netlify.app/managed-federation/deployment#using-variants-to-control-rollout)
- [Setup](https://deploy-preview-2850--apollo-federation-docs.netlify.app/managed-federation/setup#2-publish-all-subgraph-schemas)
- [Uplink](https://deploy-preview-2850--apollo-federation-docs.netlify.app/managed-federation/setup#2-publish-all-subgraph-schemas)

### CODE REVIEW

File: docs/source/managed-federation/deployment.mdx [hunk 1]
```diff
@@ -117,11 +118,95 @@ The next time it starts up or polls, your router obtains an updated configuratio
...
+    ## and retrieve the associated launch, along with any downstream launches.
+    mutation PublishSubgraphsMutation(
+      $graphId: ID!
+      $graphVariant: String!
+      $revision: String!
+      $subgraphInputs: [PublishSubgraphsSubgraphInput!]!
+    ) {
+      graph(id: $graphId) {
+        publishSubgraphs( #highlight-line
+          graphVariant: $graphVariant
```
Comment (david-castaneda) [inline docs/source/managed-federation/deployment.mdx:146]: Wanted to call out that the variant input argument is called `name` in the other queries below. Not sure if for consistency we want to use one or the other.

File: docs/source/managed-federation/deployment.mdx [hunk 2]
```diff
@@ -117,11 +118,95 @@ The next time it starts up or polls, your router obtains an updated configuratio
...
+          revision: $revision
+          subgraphInputs: $subgraphInputs
+        ) {
+          launch {
+            id
+            downstreamLaunches {
```
Comment (david-castaneda) [inline docs/source/managed-federation/deployment.mdx:152]: Since this example requests information about downstream launches we should recommend users to add as an input argument `downstreamLaunchInitiation = SYNC` ([schema docs](https://studio.apollographql.com/graph/apollo-platform/variant/main/schema/reference/enums/DownstreamLaunchInitiation#SYNC)) as part of the mutation. Otherwise these wont be created within the same request and show up as an empty list.

Comment (david-castaneda) [reply docs/source/managed-federation/deployment.mdx:152]: Given that contracts are an enterprise only feature, do we want to have a query that doesn't request downstream launches? Maybe have a separate section for querying downstream launches which shows the example with the above mentioned input argument. 

It might be confusing that the initial recommended example will have unexpected response for non-enterprise users.

Comment (david-castaneda) [reply docs/source/managed-federation/deployment.mdx:152]: Ah but maybe since this is under the "managed federation" section of the docs its completely fine.

Comment (shorgi) [reply docs/source/managed-federation/deployment.mdx:152]: Great point, will separate this as you suggest

File: docs/source/managed-federation/deployment.mdx (hunk 1 above)
Comment (shorgi) [reply docs/source/managed-federation/deployment.mdx:146]: This would need to be changed in the [publishSubgraphs](https://studio.apollographql.com/graph/apollo-platform/variant/main/schema/reference/objects/GraphMutation?_gl=1%2A1tgb7vy%2A_ga%2AOTA0MTI5NDAyLjE2OTk2NTcyMzg.%2A_ga_0BGG5V2W2K%2AMTY5OTkwNTY3Mi4yLjAuMTY5OTkwNTY3Mi4wLjAuMA..#publishSubgraphs) API itself?

Comment (david-castaneda) [reply docs/source/managed-federation/deployment.mdx:146]: Not necessarily the API can stay the same but we can change the variable names so that they can be shared between all three operations.

For example:

Variables

```json
{
  "graphId" :"my-graph",
  "graphVariant": "current"
}
```

And the operations would all use `$graphVariant` instead of `$name`

PublishSubgraphsMutation would stay as you have it

```graphql
 query PollLaunchStatusQuery($graphId: ID!, $graphVariant: String!, $launchId: ID!) {
      graph(id: $graphId) {
        variant(name: $graphVariant) {
 ...
```

```graphql
query FetchSupergraphSDLQuery($graphId: ID!, $graphVariant: String!, $launchId: String!) {
      graph(id: $graphId) {
        variant(name: $graphVariant) {
...
```

Comment (david-castaneda) [reply docs/source/managed-federation/deployment.mdx:146]: I noticed in the current state of the docs `FetchSupergraphSDLQuery` is missing the `$name`/`$graphVariant` argument in the `query`

File: docs/source/managed-federation/setup.mdx [hunk 3]
```diff
@@ -71,33 +69,37 @@ const gateway = new ApolloGateway({
 
 This option is specific to _non_-managed federation, in which supergraph schema composition is performed via the Rover CLI.
 
-With managed federation, composition is instead performed by _GraphOS_, and the gateway regularly polls for an updated schema. This enables you to add, remove, and modify your subgraphs _without_ needing to restart your gateway.
+With managed federation, composition can instead be performed by _GraphOS_, where the gateway regularly polls for an updated schema. This enables you to add, remove, and modify your subgraphs _without_ needing to restart your gateway.
 
 Remove the `supergraphSdl` argument from your `ApolloGateway` constructor entirely:
```
Comment (Meschreiber) [inline docs/source/managed-federation/setup.mdx:74]:
```suggestion
To use managed federation, ensure your `ApolloGateway` constructor doesn't include the `supergraphSdl` argument:
```

File: docs/source/managed-federation/setup.mdx [hunk 4]
```diff
@@ -71,33 +69,37 @@ const gateway = new ApolloGateway({
...
 
-If you've already set up Apollo Federation _without_ connecting your router to GraphOS, you're probably passing the `--supergraph` (or `-s`) option to the Apollo Router's startup command:
+If you've already set up Apollo Federation _without_ connecting your router to GraphOS, you're probably passing the [`--supergraph` (or `-s`) option](/router/configuration/overview/#-s----supergraph) to the Apollo Router's startup command:
 
 ```sh
 ./router --config ./router.yaml --supergraph ./your-local-supergraph.graphql
 ```
 
-The `--supergraph` option is specific to _non_-managed federation, in which supergraph schema composition is performed via the Rover CLI and provided via a file path.
+The `--supergraph` option applies in scenarios where the supergraph schema for the router isn't retrieved by polling Apollo Uplink:
```
Comment (Meschreiber) [inline docs/source/managed-federation/setup.mdx:90]:
```suggestion
The `--supergraph` option applies in scenarios where the router doesn't retrieve the supergraph schema by polling Apollo Uplink. These include:
```

File: docs/source/managed-federation/setup.mdx [hunk 5]
```diff
@@ -71,33 +69,37 @@ const gateway = new ApolloGateway({
...
+
+* For enabling different production environments (such as blue-green deployments) to refer to a single graph variant, the supergraph schema for each production environment can be composed and retrieved by a workflow using Platform APIs, so that the routers of each environment use the supergraph schema corresponding to the environment.
```
Comment (Meschreiber) [inline docs/source/managed-federation/setup.mdx:92]: I found this a little hard to follow and tried to break it up to make it more readable. That said, it seems a little contradictory to me. On the one hand, we seem to be saying that different production environments point to the same variant (and therefore the same schema?). On the other hand, we're saying each environments router can point to its (own?) corresponding supergraph schema? 

```suggestion
* **Enabling different production environments to refer to a single graph variant**, for example, for [blue-green](https://en.wikipedia.org/wiki/Blue%E2%80%93green_deployment): Workflows using the [Platform API](/graphos/platform-api/) compose and retrieve the supergraph schema for each production environment. This enables each environment's router to use the environment's supergraph schema. 
```

File: docs/source/managed-federation/setup.mdx [hunk 6]
```diff
@@ -71,33 +69,37 @@ const gateway = new ApolloGateway({
...
 ```sh
 ./router --config ./router.yaml --supergraph ./your-local-supergraph.graphql
 ```
 
-The `--supergraph` option is specific to _non_-managed federation, in which supergraph schema composition is performed via the Rover CLI and provided via a file path.
+The `--supergraph` option applies in scenarios where the supergraph schema for the router isn't retrieved by polling Apollo Uplink:
+
+* For enabling different production environments (such as blue-green deployments) to refer to a single graph variant, the supergraph schema for each production environment can be composed and retrieved by a workflow using Platform APIs, so that the routers of each environment use the supergraph schema corresponding to the environment. 
+
+* For fully unmanaged federation, the Rover CLI composes the supergraph schema, and the file path of the schema is passed to the router.
```
Comment (Meschreiber) [inline docs/source/managed-federation/setup.mdx:94]:
```suggestion
* **Unmanaged federation**: The Rover CLI composes the supergraph schema and passes the schema's file path to the router via the `--supergraph` option.
```

File: docs/source/managed-federation/setup.mdx [hunk 7]
```diff
@@ -71,33 +69,37 @@ const gateway = new ApolloGateway({
...
 
-The `--supergraph` option is specific to _non_-managed federation, in which supergraph schema composition is performed via the Rover CLI and provided via a file path.
+The `--supergraph` option applies in scenarios where the supergraph schema for the router isn't retrieved by polling Apollo Uplink:
+
+* For enabling different production environments (such as blue-green deployments) to refer to a single graph variant, the supergraph schema for each production environment can be composed and retrieved by a workflow using Platform APIs, so that the routers of each environment use the supergraph schema corresponding to the environment. 
+
+* For fully unmanaged federation, the Rover CLI composes the supergraph schema, and the file path of the schema is passed to the router. 
 
-_Remove_ the `--supergraph` option (but leave `--config` if it's present):
+For other scenarios where the supergraph schema is gotten from Apollo Uplink, _remove_ the `--supergraph` option (but leave `--config` if it's present):
```
Comment (Meschreiber) [inline docs/source/managed-federation/setup.mdx:96]:
```suggestion
For other scenarios where the router retrieves the supergraph schema is from Apollo Uplink, _remove_ the `--supergraph` option (but leave `--config` if it's present):
```

File: docs/source/managed-federation/deployment.mdx [hunk 8]
```diff
@@ -117,11 +118,96 @@ The next time it starts up or polls, your router obtains an updated configuratio
 
 ## Using variants to control rollout
 
-With managed federation, you can control which version of your graph a fleet of routers are using. In the majority of cases, rolling over all of your router instances to a new schema version is safe, assuming you've used [schema checks](./federated-schema-checks/) to confirm that your changes are backward compatible.
+Your deployment model may require deploying a specific version of your graph, for example for blue-green deployments or canary deployments. With managed federation, you can control which version of your graph a fleet of routers are using. In the majority of cases, rolling over all of your router instances to a new schema version is safe, assuming you've used [schema checks](./federated-schema-checks/) to confirm that your changes are backward compatible.
```
Comment (Meschreiber) [inline docs/source/managed-federation/deployment.mdx:121]:
```suggestion
Your deployment model may require deploying a specific version of your graph, for example, for blue-green or canary deployments. With managed federation, you can control which version of your graph your router fleet uses. In most cases, rolling over all of your router instances to a new schema version is safe, assuming you've used [schema checks](./federated-schema-checks/) to confirm that your changes are backward compatible.
```

File: docs/source/managed-federation/deployment.mdx [hunk 9]
```diff
@@ -117,11 +118,96 @@ The next time it starts up or polls, your router obtains an updated configuratio
...
 
 However, changes at the router level might involve a variety of different updates, such as [migrating entities](../entities-advanced/#migrating-entities-and-fields) from one subgraph to another. If your infrastructure requires a more advanced deployment process, you can use [graph variants](/graphos/graphs/#variants) to manage different fleets of routers running with different configurations.
```
Comment (Meschreiber) [inline docs/source/managed-federation/deployment.mdx:123]:
```suggestion
However, changes at the router level might involve a variety of different updates, such as [migrating entities](../entities-advanced/#migrating-entities-and-fields) from one subgraph to another. If your infrastructure requires a more advanced deployment process, you can use [graph variants](/graphos/graphs/#variants) to manage router fleets running with different configurations.
```

File: docs/source/managed-federation/deployment.mdx [hunk 10]
```diff
@@ -117,11 +118,96 @@ The next time it starts up or polls, your router obtains an updated configuratio
...
 
-### Example
+Specifically, for deployments that run in different production environments, such as blue-green deployments, you can configure your environments to refer to a single graph variant by pinning the supergraph schema of each environment to your routers at deployment time. Using a single variant between different production environments enables Studio to get usage reports and do analysis on the combined production traffic of all environments.
```
Comment (Meschreiber) [inline docs/source/managed-federation/deployment.mdx:125]:
```suggestion
Specifically, for deployments that run in different production environments, such as blue-green deployments, you can configure your environments to refer to a single graph variant by pinning each environment's supergraph schema to your routers at deployment time. Using a single variant between different production environments enables Studio to get usage reports and analyze the combined production traffic of all environments.
```

File: docs/source/managed-federation/deployment.mdx [hunk 11]
```diff
@@ -117,11 +118,96 @@ The next time it starts up or polls, your router obtains an updated configuratio
...
+
+### Example blue-green deployment
+
+A blue-green deployment strategy uses two environments: one environment (blue) serves the schema variant for live traffic, while the other environment (green) uses a variant for a new release that's under development. When the new release is ready, traffic is migrated from the blue to the green environment. This cycle repeats with each new release.
```
Comment (Meschreiber) [inline docs/source/managed-federation/deployment.mdx:129]:
```suggestion
A blue-green deployment strategy uses two environments: one environment (blue) serves the schema variant for live traffic. The other environment (green) uses a variant for a new release that's under development. When the new release is ready, traffic is migrated from the blue to the green environment. This cycle repeats with each new release.  
```

File: docs/source/managed-federation/deployment.mdx [hunk 12]
```diff
@@ -117,11 +118,96 @@ The next time it starts up or polls, your router obtains an updated configuratio
...
-### Example
+Specifically, for deployments that run in different production environments, such as blue-green deployments, you can configure your environments to refer to a single graph variant by pinning the supergraph schema of each environment to your routers at deployment time. Using a single variant between different production environments enables Studio to get usage reports and do analysis on the combined production traffic of all environments.
+
+### Example blue-green deployment
+
+A blue-green deployment strategy uses two environments: one environment (blue) serves the schema variant for live traffic, while the other environment (green) uses a variant for a new release that's under development. When the new release is ready, traffic is migrated from the blue to the green environment. This cycle repeats with each new release.  
+
+As an example, follow these steps to deploy with a supergraph schema of a new release (green) environment:
+
+1. Publish all subgraphs of a release at once by using the Platform API [`publishSubgraphs`](https://studio.apollographql.com/graph/apollo-platform/variant/main/schema/reference/objects/GraphMutation#publishSubgraphs).
```
Comment (Meschreiber) [inline docs/source/managed-federation/deployment.mdx:133]:
```suggestion
1. Publish all the release's subgraphs at once using the Platform API [`publishSubgraphs` mutation](https://studio.apollographql.com/graph/apollo-platform/variant/main/schema/reference/objects/GraphMutation#publishSubgraphs).
```

This link redirects to https://studio.apollographql.com/public/apollo-platform/variant/main/home for me. Is that expected?

File: docs/source/managed-federation/deployment.mdx [hunk 13]
```diff
@@ -117,11 +118,96 @@ The next time it starts up or polls, your router obtains an updated configuratio
...
+            downstreamLaunches {
+              id
+            }
+          }
+        }
+      }
+    }
+    ```
+
+  This initiates a launch, as well as any downstream launches necessary for [contracts](graphos/delivery/contracts/#automatic-updates). It returns the launch IDs, with downstream launch IDs configured to return synchronously (`downstreamLaunchInitiation: "SYNC"`) with the mutation.
```
Comment (Meschreiber) [inline docs/source/managed-federation/deployment.mdx:162]:
```suggestion
  This initiates a [launch](/graphos/delivery/launches/), as well as any downstream launches necessary for [contracts](graphos/delivery/contracts/#automatic-updates). It returns the launch IDs, with downstream launch IDs configured to return synchronously (`downstreamLaunchInitiation: "SYNC"`) with the mutation.
```

**Review by Meschreiber — COMMENTED** [review]
Left some suggestions for conciseness and clarity, otherwise LGTM!

File: docs/source/managed-federation/deployment.mdx (hunk 12 above)
Comment (shorgi) [reply docs/source/managed-federation/deployment.mdx:133]: That's unexpected. For me in a Chrome Incognito window the link opens up the Studio login page, after which it goes to the expected platform API doc.

File: docs/source/managed-federation/overview.mdx [hunk 14]
```diff
@@ -1,12 +1,15 @@
 ---
 title: Managed federation overview
+description: Let Apollo GraphOS manage validating and composing your subgraph schemas into a federated, supergraph schema and configuration for your routers.
```
[the remark this answers is not in the document]
Comment (shorgi) [reply docs/source/managed-federation/overview.mdx:3]: I like this subtitle as a copyedited description actually 😁 . Thinking of a shorter subtitle now ...

File: docs/source/managed-federation/deployment.mdx (hunk 12 above)
Comment (Meschreiber) [reply docs/source/managed-federation/deployment.mdx]: Huh, yeah. It works incognito for me, but in a browser where I'm already signed in: https://www.loom.com/share/e55bc203a82a470084bf9d05457893b5

File: docs/source/managed-federation/overview.mdx [hunk 15]
```diff
@@ -1,12 +1,15 @@
...
 
-With managed federation, your subgraphs each publish their schemas to GraphOS, which verifies that they successfully [compose](../federated-types/composition/) into a supergraph schema.
+With managed federation, you maintain subgraphs and routers while delegating GraphOS to manage the validation, composition, and update of your supergraph:
 
-On composition success, GraphOS updates your supergraph's latest configuration, which is available at a special endpoint (called the **uplink**) that your router regularly polls for updates:
+* Your subgraphs publish their schemas to GraphOS, which stores them in its schema registry, and GraphOS first [validates](./federated-schema-checks) them and then [composes](../federated-types/composition/) them into a supergraph schema. 
+
+* Your routers can poll GraphOS&mdash;specifically, its [Apollo Uplink](./uplink) endpoint&mdash;to get the latest validated supergraph schema and other configuration.
 
 ```mermaid
```
[the remark this answers is not in the document]
Comment (shorgi) [reply docs/source/managed-federation/overview.mdx:15]: `<br/>` was worse unfortunately, with `Your` getting crunched from the top. Leaving as-is for future improvement

[the remark this answers is not in the document]
Comment (Meschreiber) [reply docs/source/managed-federation/overview.mdx:15]: Hmm, on other pages, e.g. https://deploy-preview-2850--apollo-federation-docs.netlify.app/managed-federation/uplink, keeping the subgraph titles on one line does it:
image

File: docs/source/managed-federation/deployment.mdx [hunk 16]
```diff
@@ -117,11 +118,96 @@ The next time it starts up or polls, your router obtains an updated configuratio
...
+
+Specifically, for deployments that run in different production environments, such as blue-green deployments, you can configure your environments to refer to a single graph variant by pinning each environment's supergraph schema to your routers at deployment time. Using a single variant between different production environments enables Studio to get usage reports and analyze the combined production traffic of all environments.
+
+### Example blue-green deployment
+
+A blue-green deployment strategy uses two environments: one environment (blue) serves the schema variant for live traffic, and the other environment (green) uses a variant for a new release that's under development. When the new release is ready, traffic is migrated from the blue to the green environment. This cycle repeats with each new release.  
+
+As an example, follow these steps to deploy with a supergraph schema of a new release (green) environment:
```
Comment (Meschreiber) [inline docs/source/managed-federation/deployment.mdx:131]: Is some portion of this technically in preview vs. GA? If so, let's call it out.

File: docs/source/managed-federation/setup.mdx [hunk 17]
```diff
@@ -71,33 +69,37 @@ const gateway = new ApolloGateway({
 
 This option is specific to _non_-managed federation, in which supergraph schema composition is performed via the Rover CLI.
 
-With managed federation, composition is instead performed by _GraphOS_, and the gateway regularly polls for an updated schema. This enables you to add, remove, and modify your subgraphs _without_ needing to restart your gateway.
+With managed federation, composition can instead be performed by _GraphOS_, where the gateway regularly polls for an updated schema. This enables you to add, remove, and modify your subgraphs _without_ needing to restart your gateway.
```
Comment (Meschreiber) [inline docs/source/managed-federation/setup.mdx:72]: Non-blocking, just curious about the change from "is" to "can" across this page. With managed federation isn't the point that composition _is_ instead handled by GraphOS?

File: docs/source/managed-federation/setup.mdx [hunk 18]
```diff
@@ -71,33 +69,37 @@ const gateway = new ApolloGateway({
...
+If you've already set up Apollo Federation _without_ connecting your router to GraphOS, you're probably passing the [`--supergraph` (or `-s`) option](/router/configuration/overview/#-s----supergraph) to the Apollo Router's startup command:
 
 ```sh
 ./router --config ./router.yaml --supergraph ./your-local-supergraph.graphql
 ```
 
-The `--supergraph` option is specific to _non_-managed federation, in which supergraph schema composition is performed via the Rover CLI and provided via a file path.
+The `--supergraph` option applies in scenarios where the router doesn't retrieve the supergraph schema by polling Apollo Uplink. These include:
+
+* **Enabling different deployment environments to refer to a single graph variant**, for example, for [blue-green deployment](https://en.wikipedia.org/wiki/Blue%E2%80%93green_deployment): Use a workflow with the [Platform API](/graphos/platform-api/) to compose and retrieve the supergraph schema for each environment. This enables each environment's router to use the environment's supergraph schema, and then a single graph variant can capture and analyze the operations and traffic of either environment when deployed.
```
Comment (Meschreiber) [inline docs/source/managed-federation/setup.mdx:92]: Whoops, previously suggested the Wikipedia link before seeing the new section where you nicely explain the concept of blue-green deployments.

```suggestion
* **Enabling different deployment environments to refer to a single graph variant**, for example, for [blue-green deployment](./deployment/#example-blue-green-deployment): Use a workflow with the [Platform API](/graphos/platform-api/) to compose and retrieve the supergraph schema for each environment. This enables each environment's router to use the environment's supergraph schema, and then a single graph variant can capture and analyze the operations and traffic of either environment when deployed.
```

**Review by Meschreiber — APPROVED** [review]
Just wondering if we need to callout that some portion of this is in preview? Otherwise LGTM!

File: docs/source/managed-federation/overview.mdx (hunk 15 above)
[the remark this answers is not in the document]
Comment (shorgi) [reply docs/source/managed-federation/overview.mdx:15]: nice! thought the diagrams were all the same. Will use this one as a shared component

File: docs/source/managed-federation/setup.mdx (hunk 17 above)
Comment (shorgi) [reply docs/source/managed-federation/setup.mdx:72]: The distinction is that "managed federation" in general manages one or all aspects of CI/CD for federation, and composition may or may not be managed (`can` vs. `is).

That said, you're right that this sentence is focused on what GraphOS provides. Rewriting to `With federation managed by GraphOS, composition is performed and ...`

File: docs/source/managed-federation/uplink.mdx [hunk 19]
```diff
@@ -141,3 +141,19 @@ APOLLO_SCHEMA_CONFIG_DELIVERY_ENDPOINT=https://aws.uplink.api.apollographql.com/
...
+
+## Bypassing Uplink
+
+<PreviewFeature />
+
+In advanced use cases, you may want your router to use a supergraph schema different than the latest validated schema provided by Uplink. For example, you have different deployment environments for the same [graph variant](/graphos/graphs/#variants), and you want everything that managed federation provides except for your routers to use supergraph schemas specific to their deployment environment.
+
+For this scenario, you can follow a workflow that, instead of retrieving supergraph schemas from Uplink, uses the [GraphOS Platform API](/graphos/platform-api) to retrieve a supergraph schema for a specific [GraphOS launch](/graphos/delivery/launches). The workflow, in summary:
+
+1. When deploying your graphs, publish your subgraphs in a batch using a GraphOS Platform API.
```
Comment (david-castaneda) [inline docs/source/managed-federation/uplink.mdx:153]: `using a GraphOS Platform API.` -> `using the GraphOS Platform API.`?

File: docs/source/managed-federation/deployment.mdx [hunk 20]
```diff
@@ -117,11 +118,98 @@ The next time it starts up or polls, your router obtains an updated configuratio
...
+
+However, changes at the router level might involve a variety of different updates, such as [migrating entities](../entities-advanced/#migrating-entities-and-fields) from one subgraph to another. If your infrastructure requires a more advanced deployment process, you can use [graph variants](/graphos/graphs/#variants) to manage router fleets running with different configurations.
+
+Specifically, for deployments that run in different production environments, such as blue-green deployments, you can configure your environments to refer to a single graph variant by pinning each environment's supergraph schema to your routers at deployment time. Using a single variant between different production environments enables Studio to get usage reports and analyze the combined production traffic of all environments.
+
+### Example blue-green deployment
+
+<PreviewFeature />
```
Comment (david-castaneda) [inline docs/source/managed-federation/deployment.mdx:129]: I noticed this component has the following statement:
> get in touch with your Apollo contact or on the official[ Apollo GraphQL Discord](https://discord.gg/yFZJH2QYrK).

We should give a heads up to the Discord admins when this PR merges

File: docs/source/managed-federation/deployment.mdx [hunk 21]
```diff
@@ -117,11 +118,98 @@ The next time it starts up or polls, your router obtains an updated configuratio
...
+
+However, changes at the router level might involve a variety of different updates, such as [migrating entities](../entities-advanced/#migrating-entities-and-fields) from one subgraph to another. If your infrastructure requires a more advanced deployment process, you can use [graph variants](/graphos/graphs/#variants) to manage router fleets running with different configurations.
+
+Specifically, for deployments that run in different production environments, such as blue-green deployments, you can configure your environments to refer to a single graph variant by pinning each environment's supergraph schema to your routers at deployment time. Using a single variant between different production environments enables Studio to get usage reports and analyze the combined production traffic of all environments.
+
+### Example blue-green deployment
```
Comment (david-castaneda) [inline docs/source/managed-federation/deployment.mdx:127]: Do you think we should add a diagram to showcase this workflow more visually? It seems like a nice to have so not a blocker for this PR merging

File: docs/source/managed-federation/deployment.mdx [hunk 22]
```diff
@@ -117,11 +118,98 @@ The next time it starts up or polls, your router obtains an updated configuratio
...
+              }
+            }
+          }
+        }
+      }
+    }
+
+    ```
+
+1. Deploy your routers with the [`-s` or `--supergraph` option](/router/configuration/overview/#-s----supergraph) to specify the supergraph schema.
```
Comment (david-castaneda) [inline docs/source/managed-federation/deployment.mdx:206]: It wasn't clear to me that when providing this parameter the router won't poll uplink for the schema. Do you know if we call that out anywhere?

File: docs/source/managed-federation/deployment.mdx (hunk 21 above)
Comment (shorgi) [reply docs/source/managed-federation/deployment.mdx:127]: Good suggestion, adding it as a follow-up task

File: docs/source/managed-federation/deployment.mdx (hunk 22 above)
Comment (shorgi) [reply docs/source/managed-federation/deployment.mdx:206]: Good catch, adding a note here, and will update router docs too in its configuration section

File: docs/source/managed-federation/deployment.mdx [hunk 23]
```diff
@@ -117,11 +118,100 @@ The next time it starts up or polls, your router obtains an updated configuratio
...
+        publishSubgraphs( #highlight-line
+          graphVariant: $graphVariant
+          revision: $revision
+          subgraphInputs: $subgraphInputs
+          downstreamLaunchInitiation: "SYNC"
+        ) {
+          launch {
+            id
+            downstreamLaunches {
+              id
```
Comment (timbotnik) [inline docs/source/managed-federation/deployment.mdx:156]: We need to suggest that they include the `graphVariant` here since they'll need to use _this_ `graphVariant` as the input to the next 2 queries (as opposed to the base variant passed in this query)

```suggestion
              id
              graphVariant
              status
```

File: docs/source/managed-federation/deployment.mdx (hunk 1 above)
Comment (timbotnik) [reply docs/source/managed-federation/deployment.mdx:148]: IMO we should use `graphVariant` instead of `name` 👍

File: docs/source/managed-federation/deployment.mdx [hunk 24]
```diff
@@ -117,11 +118,100 @@ The next time it starts up or polls, your router obtains an updated configuratio
...
+
+### Example blue-green deployment
+
+<PreviewFeature />
+
+A blue-green deployment strategy uses two environments: one environment (blue) serves the schema variant for live traffic, and the other environment (green) uses a variant for a new release that's under development. When the new release is ready, traffic is migrated from the blue to the green environment. This cycle repeats with each new release.  
+
+As an example, follow these steps to deploy with a supergraph schema of a new release (green) environment:
+
+1. Publish all the release's subgraphs at once using the Platform API [`publishSubgraphs` mutation](https://studio.apollographql.com/graph/apollo-platform/variant/main/schema/reference/objects/GraphMutation#publishSubgraphs).
```
Comment (timbotnik) [inline docs/source/managed-federation/deployment.mdx:135]: I wonder if we need to introduce the platform API a little bit here as the link you have currently jumps straight into the weeds.

File: docs/source/managed-federation/deployment.mdx [hunk 25]
```diff
@@ -117,11 +118,100 @@ The next time it starts up or polls, your router obtains an updated configuratio
...
+            }
+          }
+        }
+      }
+    }
+    ```
+
+  This initiates a [launch](/graphos/delivery/launches/), as well as any downstream launches necessary for [contracts](graphos/delivery/contracts/#automatic-updates). It returns the launch IDs, with downstream launch IDs configured to return synchronously (`downstreamLaunchInitiation: "SYNC"`) with the mutation.
+
+1. Poll for the completed launch and any downstream launches.
```
Comment (timbotnik) [inline docs/source/managed-federation/deployment.mdx:166]: We should mention that when using contracts, the `graphVariant` used in this query needs to refer to the contract variant, which can be obtained in the previous query from `Launch.graphVariant / downstreamLaunches { graphVariant }`

File: docs/source/managed-federation/deployment.mdx [hunk 26]
```diff
@@ -117,11 +118,100 @@ The next time it starts up or polls, your router obtains an updated configuratio
...
+            status
+          }
+        }
+      }
+    }
+
+    ```
 
-However, changes at the router level might involve a variety of different updates, such as [migrating entities](../entities-advanced/#migrating-entities-and-fields) from one subgraph to another. If your infrastructure requires a more advanced deployment process, you can use [graph variants](/graphos/graphs/#variants) to manage different fleets of routers running with different configurations.
+1. After the launch and downstream launches have completed, retrieve the supergraph schema of the launch.
```
Comment (timbotnik) [inline docs/source/managed-federation/deployment.mdx:182]: We should again mention that when using contracts, the `graphVariant` used in this query needs to refer to the contract variant, which can be obtained in the previous query from `Launch.graphVariant / downstreamLaunches { graphVariant }`

File: docs/source/managed-federation/deployment.mdx [hunk 27]
```diff
@@ -115,15 +116,128 @@ rover subgraph delete my-supergraph@my-variant --name accounts
...
+        }
+      }
+    }
+    ```
+
+  This initiates a [launch](/graphos/delivery/launches/), as well as any downstream launches necessary for [contracts](graphos/delivery/contracts/#automatic-updates). It returns the launch IDs, with downstream launch IDs configured to return synchronously (`downstreamLaunchInitiation: "SYNC"`) with the mutation.
+
+  <Note>
+
+  For contracts, the contract variants are gotten from this query with ` Launch.graphVariant / downstreamLaunches { graphVariant }`. These are used in queries in following steps.
```
Comment (timbotnik) [inline docs/source/managed-federation/deployment.mdx:174]:
```suggestion
  For contracts, you can also request that any downstream launches return the variant associated with each launch,  e.g. `downstreamLaunches { graphVariant }`.  When querying for a specific launch, be sure to pass the variant associated with the launch in the following steps.
```

File: docs/source/managed-federation/deployment.mdx [hunk 28]
```diff
@@ -115,15 +116,128 @@ rover subgraph delete my-supergraph@my-variant --name accounts
...
+          }
+        }
+      }
+    }
+
+    ```
+
+    <Note>
+
+    When polling for a contract, the `$graphVariant` argument of this query must refer to a contract variant. You can get it from the query in step 1, from `Launch.graphVariant / downstreamLaunches { graphVariant }`.
```
Comment (timbotnik) [inline docs/source/managed-federation/deployment.mdx:196]:
```suggestion
    When polling for a contract, the `$graphVariant` argument of this query must refer to the contract variant rather than the base variant. You can get it from the query in step 1, from `Launch.graphVariant / downstreamLaunches { graphVariant }`.
```

**Review by timbotnik — APPROVED** [review]
Left one more suggestion but IMO we are good here 👏

**Review by david-castaneda — APPROVED** [review]
